File: app/src/main/java/io/jeffchang/buttondemo/ui/userlist/view/UserListRecyclerViewAdapter.java
package io.jeffchang.buttondemo.ui.userlist.view;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RelativeLayout;

import java.util.ArrayList;
import java.util.List;

import io.jeffchang.buttondemo.models.User;

/**
 * RecyclerView Adapter for dog breed list.
 */

public class UserListRecyclerViewAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private Context context;
    private ArrayList<User> users;

    UserListRecyclerViewAdapter(Context context, List<User> users) {
        this.context = context;
        this.users = new ArrayList<>(users);
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ViewGroup.LayoutParams layoutParams = new RelativeLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.MATCH_PARENT);
        UserListItem userListItem = new UserListItem(context);
        userListItem.setLayoutParams(layoutParams);
        return new UserViewHolder(userListItem);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {

    }

    @Override
    public int getItemCount() {
        return users.size();
    }

    class UserViewHolder extends RecyclerView.ViewHolder {

         UserViewHolder(View itemView) {
            super(itemView);
        }
    }
}

File: app/src/main/java/io/jeffchang/buttondemo/ui/userlist/view/UserListRecyclerViewAdapter.kt
package io.jeffchang.buttondemo.ui.userlist.view

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.ViewGroup

import java.util.ArrayList

import io.jeffchang.buttondemo.models.User

/**
 * RecyclerView Adapter for dog breed list.
 */

class UserListRecyclerViewAdapter constructor(
        private val context: Context,
        users: List<User>) : RecyclerView.Adapter<UserListRecyclerViewAdapter.UserViewHolder>() {
    private val users: ArrayList<User> = ArrayList(users)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UserViewHolder
        = UserViewHolder(UserListItem(context))

    override fun onBindViewHolder(holder: UserViewHolder, position: Int) {
        holder.userListItem.name = users[position].name
        holder.userListItem.email = users[position].email
    }

    override fun getItemCount() = users.size

    inner class UserViewHolder(val userListItem: UserListItem)
        : RecyclerView.ViewHolder(userListItem)
}
